0.6.1/docker-entrypoint.sh

```diff
@@ -61,4 +61,9 @@ elif vault --help "$1" 2>&1 | grep -q "vault $1"; then
     set -- vault "$@"
 fi
 
+# If we are running Vault, make sure it executes as the proper user.
+if [ "$1" = 'vault' ]; then
```

Comment:
The problem is that your chown is in the Dockerfile, while in the Consul container I linked to it's in the entrypoint. So it happens at runtime *after* the bind mount to fix files that are bind mounted in.